repair/row_level.cc

```diff
@@ -2448,6 +2450,34 @@ future<> repair_service::init_ms_handlers() {
                 dht::token::to_int64(range_start), dht::token::to_int64(range_end)).discard_result();
         co_return repair_update_system_table_response();
     });
+    ms.register_repair_flush_hints_batchlog([this] (const rpc::client_info& cinfo, repair_flush_hints_batchlog_request req) -> future<repair_flush_hints_batchlog_response> {
+        auto from = cinfo.retrieve_auxiliary<gms::inet_address>("baddr");
+        rlogger.info("repair[{}]: Started to process repair_flush_hints_batchlog_request from node={}, target_nodes={}, hints_timeout_in_seconds={}, batchlog_timeout_in_seconds={}",
+                req.repair_uuid, from, req.target_nodes, req.hints_timeout_in_seconds, req.batchlog_timeout_in_seconds);
+        std::vector<gms::inet_address> target_nodes(req.target_nodes.begin(), req.target_nodes.end());
+        db::hints::sync_point sync_point = co_await _sp.local().create_hint_sync_point(std::move(target_nodes));
+        lowres_clock::time_point deadline = lowres_clock::now() + std::chrono::seconds(req.hints_timeout_in_seconds);
+        try {
+            rlogger.info("repair[{}]: Started to flush hints for repair_flush_hints_batchlog_request from node={}, target_nodes={}", req.repair_uuid, from, req.target_nodes);
+            co_await _sp.local().wait_for_hint_sync_point(std::move(sync_point), deadline);
+            rlogger.info("repair[{}]: Finished to flush hints for repair_flush_hints_batchlog_request from node={}, target_hosts={}", req.repair_uuid, from, req.target_nodes);
+
+            rlogger.info("repair[{}]: Started to flush batchlog for repair_flush_hints_batchlog_request from node={}, target_nodes={}", req.repair_uuid, from, req.target_nodes);
+            co_await _bm.local().do_batch_log_replay();
+            rlogger.info("repair[{}]: Finished to flush batchlog for repair_flush_hints_batchlog_request from node={}, target_nodes={}", req.repair_uuid, from, req.target_nodes);
```

Comment:
We might do these in parallel (see coroutine::all()).

Reply:
I assume you mean flush hint and batchlog in parallel? Thanks for the coroutine::all() tip. Actually, I think we should flush batchlog then flush hints. batchlog may write hints in the future. Current, our batchlog replay does not use hint though. 

Thanks for the coroutine::all() tip.

Reply:
Done. It is fun to use coroutine::all().